plugin-server/src/worker/ingestion/ingest-event.ts

```diff
@@ -67,9 +68,9 @@ export async function ingestEvent(hub: Hub, event: PluginEvent): Promise<IngestE
 }
 
 export async function ingestBufferEvent(hub: Hub, event: PreIngestionEvent): Promise<IngestEventResponse> {
-    const person = await hub.db.fetchPerson(event.teamId, event.distinctId)
+    const person = await hub.db.getPersonData(event.teamId, event.distinctId)
```

Comment:
I'm a bit worried about this, because we haven't processed the person properties that came with this event, i.e. which might change action matching or cohort belonging?

Reply:
Yes we have - we do all the event processing and dump it into the buffer only to be ingested later with the updated person data.

Reply:
Right, sorry I forgot the ordering, then we have a different problem - specifically the event might now be saved with different person properties compared to what were sent with this event. Given that our suggestions for person property problems was send them with the event they need to be on. So we might want to do the person properties processing later. 
Something like:
capture(event) -> handle Alias & Identify (but don't add the properties in this event yet) -> split to ingestion / buffer -> ingest&buffer both: apply person properties 

Note this will make apply props more out of sync (by buffer length), but again we're using the assumption that person props shouldn't change often). I think the properties sent in the event are more important to match the event compared to ordering differences.
We can also decide it's not important and just ship forward, but this is likely going to cause some user questions later.

Reply:
that is a valid point.

I'm unsure what the best approach is. I was about to agree with doing processing later but actually not sure that's the best approach.

With that we'd see some props flip at random times, consider:

```
event, ts, person_props
-----------
event0, 0, { a: 1 }
event1, 65, { a: 2 }
event3, 70, { }
```

`event0` goes to the buffer, meanwhile `event1` does not. Say event0 gets processed at t=70, now the props going forward will be { a: 1 } instead of { a: 2 }.

I think we should process these immediately (cause we'll also be guaranteeing order to some extent), but what we can do is when processing the buffer event we do:

```js
const personProps = { ...currentPersonProps, ...personPropsInEvent }
```

That makes sure we have up to date props but reflecting the updates that event had as well.

How does this sound?

Reply:
`const personProps = { ...currentPersonProps, ...personPropsInEvent }`

seems fine to me & group props don't have this problem.